### PR TITLE
Fast Track Next.js

### PR DESCRIPTION
**Pull request by acald-creator**
**Notes for Reviewers**

This PR is related to https://github.com/meshery/meshery/issues/6680 and https://github.com/meshery/meshery/pull/6747.

This is simply to fast track Next.js to `v12` and React to `v17`.

Unfortunately I ran into some styling issues at Next.js `13` and React `18` which will require a review of the current Material UI v4 to be migrated to Material UI v5 sooner than later.

A few things to note:

`eslint` is pinned at v7 for now because of an issue with Next.js 12 which is looking for eslint to be installed as a dev dependencies but the message didn't go away until I pinned it at this version.

I removed some of the `eslint` plugins and config because they either have been merged together or deprecated in favor of something else, but in this PR, most of them have been merged into one, where Next.js has better built-in support for eslint rules. Eslint will be a bit noisy, so we will have to review and decide which to turn off.

Because we are upgrading to React 17-18, any new React Components will now need to start using React Function Components instead of React Class Components, and this also means using React Hooks.

Lastly, Cypress had some dependencies issues, so I upgraded webpack to v5 and installed ajv in the meantime. In my local development, the test seemed to be passing, but we will leave it others to provide feedback and test this PR.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -235,7 +235,7 @@ ui-build:
 
 ## Build only Meshery UI on your local machine.
 ui-meshery-build:
-	cd ui; npm run $(UI_BUILD_SCRIPT) && npm run export; cd ..
+	cd ui; npm run build && npm run export; cd ..
```
Comment (Abhishek-kumar09) [inline Makefile:238]: Why this change?

File: ui/.npmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+legacy-peer-deps=true
```
[the remark this answers is not in the document]
Comment (acald-creator) [reply ui/.npmrc:1]: `react-codemirror2` and `mui v4` are the only real blockers.

There are other smaller packages but those are already noted on my other PR.

File: install/docker/Dockerfile [hunk 3]
```diff
@@ -11,7 +11,7 @@ RUN go clean -modcache; cd server; cd cmd; GOPROXY=https://proxy.golang.org GOSU
 
 FROM node:lts as ui
 ADD ui ui
-RUN cd ui; npm install --only=production; npm run build && npm run export; mv out /
+RUN cd ui; npm install --omit-dev; npm run build && npm run export; mv out /
```
[the remark this answers is not in the document]
Comment (acald-creator) [reply install/docker/Dockerfile:14]: When I was running docker-build, I saw the message regarding this particular line was deprecated, node:lts is node 18

File: Makefile (hunk 1 above)
Comment (acald-creator) [reply Makefile:238]: For the purpose of testing Cypress

**Review by leecalcote — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (Abhishek-kumar09) [reply Makefile:238]: I dont think this is affecting cypress at all. Now a new problem arises, that will stop us from building meshery with any node version later than 16
